src/braket/circuits/angled_gate.py

```diff
@@ -41,22 +48,32 @@ def __init__(self, angle: float, qubit_count: Optional[int], ascii_symbols: Sequ
         super().__init__(qubit_count=qubit_count, ascii_symbols=ascii_symbols)
         if angle is None:
             raise ValueError("angle must not be None")
-        self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+        if isinstance(angle, FreeParameter):
+            self._angle = angle
+            self._parameterized = True
+            self._parameter = angle
+        else:
+            self._angle = float(angle)  # explicit casting in case angle is e.g. np.float32
+            self._parameterized = False
+            self._parameter = None
 
     @property
-    def angle(self) -> float:
+    def angle(self) -> Union[FreeParameter, float]:
         """
         Returns the angle for the gate
 
         Returns:
-            angle (float): The angle of the gate in radians
+            angle (Union[FreeParameter, float]): The angle of the gate in radians
         """
         return self._angle
 
     def __eq__(self, other):
```

Comment:
Can you describe in a comment what equality does here?
Also, is this really what we want?

Reply:
This checks the angled gate is equal based on the angle based upon our previous definition. As the angle can be a float or free parameter, equality is checked for the two branches.

Reply:
what if self.angle is a float but other.angle is a FreeParameter?

Reply:
```
self.angle == other.angle
```
Would run a check of a free parameter against a float here. FP has a check of the type for the equality operator so it would be false in this case.

Reply:
And this is true even if the free parameter has been set to the same value as the float?

Reply:
Yes FreeParameter equality is checked based on the name and the instance type. The check would return false as the comparison would be between a FreeParameter and a float

Reply:
Wanted to stress again that many users are accustomed to SymPy - what's the behavior there? I'm hoping anything we do here will not be too surprising.

Reply:
Yeah, while I agree that we shouldn't do stuff that seems weird for the customers, we don't want to be restricted either if what SymPy does isn't right for us. However, I completely agree that we should understand what SymPy does and make a conscious decision about what path we'll take.